packages/react-menu/src/components/MenuList/index.ts

```diff
@@ -1,5 +1,14 @@
 export * from './MenuList';
-export * from './MenuList.types';
+// Explicit exports to ommit MenuListCommon
+export type {
+  MenuCheckedValueChangeEvent,
+  MenuCheckedValueChangeData,
+  MenuListSlots,
+  MenuListProps,
+  MenuListState,
+  MenuListContextValues,
+  UninitializedMenuListState,
+} from './MenuList.types';
```

Comment:
nit: Order these alphabetically
```suggestion
export type {
  MenuCheckedValueChangeData,
  MenuCheckedValueChangeEvent,
  MenuListContextValues,
  MenuListProps
  MenuListSlots,
  MenuListState,
  UninitializedMenuListState,
} from './MenuList.types';
```